tests/backends/launcher.py

```diff
@@ -0,0 +1,168 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import sys
+import os
+import coverage
+import subprocess
+from subprocess import TimeoutExpired
+from pathlib import Path
+import pytorch_lightning
+from argparse import ArgumentParser
+from pytorch_lightning import Trainer, seed_everything
+from tests.base import EvalModelTemplate
+import torch
+import functools
+import itertools
+from time import time
+from inspect import isfunction, ismethod, isclass
+
+
+def import_from(module, name):
+    module = __import__(module, fromlist=[name])
+    return getattr(module, name)
+
+
+def call_training_script(cli_args, tmpdir, env, timeout=20):
+    file = Path(__file__).absolute()
+    cli_args = cli_args.split(' ') if cli_args else []
+    cli_args += ['--tmpdir', str(tmpdir)]
+    command = [sys.executable, '-m', 'coverage', 'run', str(file)] + cli_args
+
+    # need to set the PYTHONPATH in case pytorch_lightning was not installed into the environment
+    env['PYTHONPATH'] = f'{pytorch_lightning.__file__}:' + env.get('PYTHONPATH', '')
+
+    # for running in ddp mode, we need to lauch it's own process or pytest will get stuck
+    p = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=env)
+
+    try:
+        std, err = p.communicate(timeout=timeout)
+        err = str(err.decode("utf-8"))
+        if 'Exception' in err:
+            raise Exception(err)
+    except TimeoutExpired:
+        p.kill()
+        std, err = p.communicate()
+    return std, err
+
+
+def create_cmd_lines(cmd_line, **kwargs):
+    keys = kwargs.keys()
+    keys = sorted(keys)
```

Comment:
```suggestion
    keys = sorted(kwargs.keys())
```
what is the reason for being sorted?